### PR TITLE
Decompiled 2 functions in iMath3.cpp as well as partaily iSphereIsectRay

### PR DESCRIPTION
**Pull request by sonich2401**
Decompiled 2 Functions as well as partailly iSphereIsectRay. Also in progress to decompile Vec3Copy

### CODE REVIEW

File: src/Core/p2/iMath3.cpp
Comment (mattbruv) [inline src/Core/p2/iMath3.cpp:8]: Make sure you format every file you commit with clang-format with our ruleset

File: .gitignore
Comment (mattbruv) [inline .gitignore]: I don't know if a program generated this stuff or if you handwrote it, but none of this should be here. A lot of those things are scripts that we do want tracked by git if they are changed.

File: src/Core/p2/iMath3.cpp
Comment (mattbruv) [inline src/Core/p2/iMath3.cpp:15]: use our custom typedefs: i.e. `float32` instead of `float`

File: src/Core/p2/iMath3.cpp
Comment (mattbruv) [inline src/Core/p2/iMath3.cpp:110]: I get that this  probably is close to matching, but it's pretty useless as far as readable code goes. If you compare this function to `iSphereIsectVec`, it is much more legible and correct than this. This looks like you basically copied and pasted ghidra's output into a comment, which doesn't really help much honestly. If it almost matches and you want to keep it, this function needs to be cleaned up A LOT. Use the struct definitions to set the properties of a type, not random pointer math everywhere. Basically make this look like `iSphereIsectVec`, and use our custom types everywhere, and format the code with clang-format before I can accept this. Alternatively, you can submit the two other functions in a separate PR and I can merge them (after other issues are addressed), but I don't want to merge this code in

File: src/Core/p2/iMath3.h
Comment (mattbruv) [inline src/Core/p2/iMath3.h]: You should include `<types.h>` instead of re-defining our types

File: src/Core/x/xVec3.cpp
Comment (mattbruv) [inline src/Core/x/xVec3.cpp:18]: This doesn't 🆗 right? I can't merge this code if it's uncommented and doesn't build properly. Also, as discussed in discord, this function was probably hand-written assembly, so I highly doubt that's anything close to what they wrote

**Review by mattbruv — CHANGES_REQUESTED** [review]

File: src/Core/p2/iMath3.cpp
Comment (sonich2401) [reply src/Core/p2/iMath3.cpp:110]: Yea I was on 0 sleep when I made this PR. Tbh i shouldn't have made this PR yet. How about I just delete this PR until I have my bearings straight?

**Comment by sonich2401** [comment]
Closed Until I can get my act together and learn how to make a proper PR lol.

**Pull request closed without merging**
